next/components/forms/widget-components/Upload/Upload.tsx

```diff
@@ -129,30 +201,30 @@ const UploadComponent: ForwardRefRenderFunction<HTMLDivElement, UploadProps> = (
     return chosenFiles
   }
 
-  const addNewFiles = (newFiles: UploadMinioFile[]) => {
+  const addNewFiles = async (newFiles: UploadMinioFile[]) => {
     const sanitizedFiles = sanitizeClientFiles(newFiles)
-    if (multiple && value && value[0]) {
+    const oldFiles = value ? [...value] : []
+    if (multiple && oldFiles.length > 0 && oldFiles[0]) {
       removeFirstFile()
     }
```

Comment:
I don't quite understand why this happens - why just the first file is removed, only when multiple files can be uploaded - isn't the condition switched around ? shouldn't this happen when it's `!multiple` (the logic being: if only one file can be uploaded uploading new file removes the old one, otherwise both old and new are kept) ?